adding simple docs site

choldgraf opened the pull request:
I got tired of being unable to preview the docs locally and also waiting for circleci to build our live preview, so this PR adds a simple "demo docs" site and uses that for the live preview instead of the full pandas documentation.

Right now, it's mostly a copypasta from the [readthedocs theme docs](https://github.com/readthedocs/sphinx_rtd_theme). Importantly, that has a bunch of sections that cover different structural / page elements. I haven't modified anything yet, because I think a reasonable first goal should be "the pandas sphinx theme should produce a good-looking website using the readthedocs theme site".

I didn't delete the pandas docs here, because I wasn't sure whether or not it was important to keep it. Happy to do that if you think it's fine.

What do you think?

choldgraf commented:
Here's the demo link:

https://30-130237506-gh.circle-artifacts.com/0/html/index.html

It has a lot of structural elements missing but I think those will be fixed once #50 lands

FWIW, the docs preview build on Circle now takes about 30 seconds compared with ~9 minutes before :-)

jorisvandenbossche commented:
Cool, thanks!

Some elements that are important for pandas to have in the demo site:

- rendering of a numpydoc docstring (but it doesn't need to be that many as now, a single one should be enough :-))
- autosummary tables (we had some problems with rendering of those, and some specific fixes for it)
- 2 levels in the left sidebar

Some elements that are not important to keep:

- the huge amount of pages ;)
- the ipython directive usage (in the end this just produces code-blocks, so shouldn't impact the styling itself)

There might also be some content to take from http://ryan-roemer.github.io/sphinx-bootstrap-theme/README.html

choldgraf commented:
Thanks for the feedback, I just added the following:

* Make it so that the top navbar items are much less-cluttered, now there's just a "demo" section at the top.
* Add an `autosummary` and `automodule` demo for some Pandas modules to the "API" section of the demo site.
* Rebased on master so we can preview the site :-)

How does that look?

jorisvandenbossche commented:
I removed some irrelevant content (eg how to configure the rtd theme, this is not applicable for us (yet)).

And it builds blazingly fast now here. While I am typing his comment it already finished and updated the link ;)

choldgraf commented:
Nice! That sounds great to me.

For this PR, do you think we should also remove the pandas documentation from the repository? Or do you need that in there for testing etc? Either way, I'm happy to merge whenever you're happy

jorisvandenbossche commented:
Let's keep the pandas docs one for a bit longer, it makes it a bit easier for us to experiment with something pandas specific. But so you can ignore them :)

Thanks for the PR!
